mipengine/node_tasks_DTOs.py

```diff
@@ -53,12 +54,59 @@ class ColumnInfo(ImmutableBaseModel):
 class TableSchema(ImmutableBaseModel):
     columns: List[ColumnInfo]
 
+    @property
+    def column_names(self):
+        return [column_info.name for column_info in self.columns]
+
 
 class TableInfo(ImmutableBaseModel):
     name: str
     schema_: TableSchema
     type_: TableType
 
+    @property
+    def column_names(self):
+        return self.schema_.column_names
+
+    @property
+    def tablename_parts(self) -> Tuple[str, str, str, str]:
+        table_type, node_id, context_id, command_id, command_subid = self.name.split(
+            "_"
+        )
+        return node_id, context_id, command_id, command_subid
+
+    @property
+    def node_id(self) -> str:
+        node_id, _, _, _ = self.tablename_parts
```

Comment:
Shorter version node_id, *_  = self.tablename_parts

Reply:
You are right but I prefer it to be similar to the rest.

Reply:
These are not the same actually! `a, _, _, _` expects exactly 4 items and raises an error otherwise. `a, *_` expects at least one item. So the first one is a bit more defensive.